### PR TITLE
wapiti: init at 3.0.4

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Wapiti allows you to audit the security of your websites or web applications.

It performs "black-box" scans (it does not study the source code) of the web application by crawling the webpages of the deployed webapp, looking for scripts and forms where it can inject data.

Once it gets the list of URLs, forms and their inputs, Wapiti acts like a fuzzer, injecting payloads to see if a script is vulnerable.

https://wapiti-scanner.github.io/

`yaswfp` is required. This module was not updated for some time now and this could be a blocker.

Related to #81418

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 117984` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python38Packages.yaswfp</li>
    <li>python39Packages.yaswfp</li>
    <li>wapiti</li>
  </ul>
</details>

File: pkgs/development/python-modules/yaswfp/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+buildPythonPackage rec {
+  pname = "yaswfp";
+  version = "0.9.3";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "138w6pnn75vd3ivykc7wsafhg1izjvq9bzz1398vnss7cxrhawb1";
+  };
+
+  # Tests are only available in repo
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/yaswfp/default.nix:15]: Can we fetch that repo?

File: pkgs/tools/security/wapiti/default.nix [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  propagatedBuildInputs = with python3.pkgs; [
+    requests
+    beautifulsoup4
+    tld
+    yaswfp
+    Mako
+    pysocks
+    markupsafe
+    six
+    importlib-metadata
```
Comment (SuperSandro2000) [inline pkgs/tools/security/wapiti/default.nix:30]: This shouldn't be required in 3.9 anymore.

File: pkgs/development/python-modules/yaswfp/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/yaswfp/default.nix:15]: We can and running the tests is possible.

File: pkgs/tools/security/wapiti/default.nix (hunk 2 above)
Comment (fabaff) [reply pkgs/tools/security/wapiti/default.nix:30]: Thanks, that slipped in.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 117984` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python38Packages.yaswfp</li>
    <li>python39Packages.yaswfp</li>
    <li>wapiti</li>
  </ul>
</details>

**Pull request merged**